### ISSUE TITLE
[Re-Introspection] `@@map` and `@map` with argument `name` loses the `name` after re-introspection

### ISSUE THREAD

**Issue by divyenduz** [prisma/prisma#3053]
`@@map` with `name` (optional argument label) in schema.prisma, when re-introspected drops the optional argument label. 

Modified schema: 

```prisma
model User {
  id Int @id
  name String
  type UserType @default(NORMAL)

  @@map(name: "user")
}
```

Re-introspected schema: 

```prisma
model User {
  id Int @id
  name String
  type UserType @default(NORMAL)

  @@map("user")
}
```

Diff: 

```diff
-   @@map(name: "user")
+   @@map("user") 
```

Our docs use the `@@map(name: value)` variant

**Comment by do4gr** [comment]
As keeping the `name` would necessitate major changes and does not really provide value I would prefer us to just drop the `name` from our examples and docs.

**Comment by divyenduz** [comment]
Marked it as a candidate so it gets brought up in the planning meeting. Candidate because it is a bit confusing as a user.

**Comment by matthewmueller** [comment]
I'm +1 to this change. The first argument in our attributes has is optionally unkeyed.

**Comment by janpio** [comment]
What change exactly @matthewmueller? Dropping it from docs and examples as @do4gr suggested or making sure to keep it in Re-Introspection if it was present in the original schema as @divyenduz suggested (or both)?

**Comment by matthewmueller** [comment]
Sorry, dropping the `name` from the docs.

**Comment by janpio** [comment]
What about the Re-Introspection behavior though? Should we always default to drop the `name` although it is supported and allowed in PSL? Then this issue can probably be closed as no change in Re-Introspection is necessary.

**Comment by matthewmueller** [comment]
Right... I don't have much of an opinion on this. If they already have a `@map(name:"...")`, intuitively it would probably make sense to maintain it during re-introspection. If we're removing it, it's a bit like re-formatting, which I don't think is the goal of re-introspection.

It's such a small thing though that if it's hard to pull off in re-introspection, I don't think it matters much.

My suggestion: close it and if people complain we can revisit.

**Comment by janpio** [comment]
Context: @divyenduz opened this issue as it screws up diffs in Re-Introspection-CI.

**Comment by matthewmueller** [comment]
Gotcha. 

- How painful is this @divyenduz? Is it a one-time cost? Or do you constantly bump into it?
- @do4gr how much work would this be to fix it for @divyenduz?

**Comment by mavilein** [comment]
It would be a lot of work to fix this and it would probably complicate one of our core abstractions significantly.

**Comment by divyenduz** [comment]
> how painful is this? Is it a one-time cost? Or do you constantly bump into it?

I only ran into it because I got the `name` syntax from the docs, otherwise, this should not be an issue. If possible, we can consider removing the `name` label altogether but that would, of course, be a breaking change. 

When `name` label is used, it makes re-introspection confusing to understand though, my first impression was as if re-introspection didn't work.
